LanguageFeatures/Unquoted-imports/path_separator_A01_t01.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Instead of `.`, this proposal uses `/`. This is obviously a
+/// natural path separator since it's the main path separator character used in
+/// most operating systems and in URLs on the web. Also, it is already the path
+/// separator character used inside Dart imports today.
+///
+/// @description Checks that it is a compile-time error to use `.` or `\` as a
+/// path separator.
```

Comment:
I don't think it is possible to maintain that any particular piece of text is "an unquoted Dart import directive that uses `X` as a path separator" where `X` is some character. So we can't really say that we're testing that it would be an error to do so.

The problem is that every character which is accepted by the lexical rules plays a particular role, and there's no way we can "use it as" anything other. In particular, `.` and `\` are never path separators.

So what we _can_ do is to test that the unquoted path `test.package1.dotted.name1` is transformed into the URI `package:test.package1.dotted.name1/name1.dart`. We would make sure that there is a package like this, and perhaps read a `libraryId` to confirm that we're importing the library that we expect to import.

We could also arrange for a package named `test` to exist, and for that package to have a `lib` that contains `package1/dotted/name1.dart`, and then we'd be able confirm that the import directive does _not_ import that library (that is, we don't get that value of `libraryId`), but that wouldn't really show anything if we already have confirmed that we're importing the right library (it's fair to assume that the same import directive won't import two or more libraries; besides, if we're checking `libraryId` then we _aren't_ importing both libraries because then `libraryId` would be a conflicted name ;-).

Reply:
Ok, let's remove `path_separator_*` tests then. I'll `static_semantics_*` tests in a separate PR that will proof that a path with `.` and `/` is transformed and points to a right library. This PR only adds necessary packages for the furter tests then.

Reply:
Sounds good, thanks!